kusto/kcsb.go

```diff
@@ -163,6 +175,35 @@ func (kcsb *ConnectionStringBuilder) resetConnectionString() {
 	kcsb.RedirectURL = ""
 	kcsb.ClientOptions = nil
 	kcsb.DefaultAuth = false
+
+	kcsb.setDefaultTracingValues()
+}
+
+func (kcsb *ConnectionStringBuilder) setDefaultTracingValues() {
+	kcsb.ApplicationForTracing = filepath.Base(os.Args[0])
+	kcsb.UserForTracing = getOsUser()
+	kcsb.versionForTracing = buildHeaderFormat(StringPair{Key: "Kusto.Go.Client", Value: version.Kusto}, StringPair{Key: "Go", Value: runtime.Version()})
+}
+
+func getOsUser() string {
+	var final string
+	current, err := user.Current()
+	if err != nil || current.Username == "" {
```

Comment:
nit: this is odd cause you used the checked value in the else
Reverse the condition perhaps?
You can also consider scope reduction (if {return} X N)